src/NServiceBus.Core/Sagas/Sagas.cs

```diff
@@ -58,13 +58,13 @@ protected internal override void Setup(FeatureConfigurationContext context)
                 sagaMetaModel.VerifyIfEntitiesAreShared();
             }
 
-            RegisterCustomFindersInContainer(context.Container, sagaMetaModel);
+            RegisterCustomFindersInContainer(context.Services, sagaMetaModel);
 
             foreach (var t in context.Settings.GetAvailableTypes())
             {
                 if (IsSagaNotFoundHandler(t))
                 {
-                    context.Container.ConfigureComponent(t, DependencyLifecycle.InstancePerCall);
+                    context.Services.AddWithInterfaces(t, ServiceLifetime.Transient);
```

Comment:
This one breaks reolution of saga not found handlers but changing to

```
context.Services.AddTransient(typeof(IHandleSagaNotFound), t);
```

makes all tests pass again